### PR TITLE
vamp signature updates

### PR DESCRIPTION
**Pull request by rakuy0**
* Adds a small test for the envi/bytesig.py code
* In the case where you had signatures A, B, and C, where A is a subset of B, and B is a subset of C, you'd get various IndexErrors when either adding them to the know sigs or matching against them.
* Fix a display issue in the README
* For the genSigAndMask, allow for making signatures in the middle of a function (to support cases where the first code block is really just a jump to the rest of the function)

### CODE REVIEW

File: vivisect/vamp/__init__.py [hunk 1]
```diff
@@ -42,10 +42,13 @@ def genSigAndMask(vw, funcva):
...
-
-    # Figgure out the size of the first linear chunk
+    if startva == None:
+        startva = funcva
+    # Figure out the size of the first linear chunk
     # in this function...
-    cb = vw.getCodeBlock(funcva)
+    cb = vw.getCodeBlock(startva)
+    if cb[2] != funcva:
+        raise Exception("startva not in given func")
```
Comment (atlas0fd00m) [inline vivisect/vamp/__init__.py]: this is a trap.  Viv currently only tracks one possible function for each codeblock, although in reality, this is not always so.  if that codeblock is the function start, this isn't a problem.  however, if it's not, this becomes a problem.
iinm, currently the only accurate way to access the function's codeblocks:

vw.getFunctionBlocks(funcva)

something like
if cb[CB_VA] not in [cbva for cbva,cbsz,cbfva in vw.getFunctionBlocks(funcva)]:

or something more elegant... but equally as accurate.

File: envi/bytesig.py [hunk 2]
```diff
@@ -33,22 +35,27 @@ def _addChoice(self, siginfo, node):
         while len(todo):
 
             node, siginfo = todo.pop()
-            depth, sigs, choices = node
+            depth, sigs, choices, term = node
             bytes, masks, o = siginfo
             siglen = len(sigs)
-
-            sigs.append(siginfo)
+            if len(bytes) > depth:
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/bytesig.py]: Fair. I was trying to follow the established convention. I'm not a fan of it either, but I do want it all to look (relatively) the same.

File: envi/bytesig.py [hunk 3]
```diff
@@ -33,22 +35,27 @@ def _addChoice(self, siginfo, node):
...
+                sigs.append(siginfo)
+            else:
+                term.append(siginfo)
+                continue
+            # If one sig is [85, 139, 236] and another is [85, 139, 236, 232, 144], then
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/bytesig.py]: :+1:

File: vivisect/vamp/__init__.py (hunk 1 above)
Comment (rakuy0) [reply vivisect/vamp/__init__.py]: Good catch. I'll change that around.

**Pull request merged**